receiver/kubeletstatsreceiver/runnable.go

```diff
@@ -57,19 +60,23 @@ func (r *runnable) Setup() error {
 }
 
 func (r *runnable) Run() error {
-	const dataformat = "kubelet"
 	const transport = "http"
-	ctx := obsreport.StartMetricsReceiveOp(r.ctx, dataformat, transport)
+	ctx := obsreport.ReceiverContext(r.ctx, typeStr, transport, r.receiverName)
+	ctx = obsreport.StartMetricsReceiveOp(ctx, typeStr, transport)
```

Comment:
Ifaik `StartMetricsReceiveOp` calls must always match with `EndMetricsReceiveOp`. So we shouldn't put `EndMetricsReceiveOp` under a condition. I don't think we need `StartMetricsReceiveOp` call here. We probably only need only one `StartMetricsReceiveOp` call and one `EndMetricsReceiveOp` call in this function, both of them should wrap `err = r.consumer.ConsumeMetricsData(ctx, *md)` in every iteration of the loop over MetricsData.

Reply:
Agreed, will fix.